lib/api/core/models/security/role.js

```diff
@@ -53,10 +53,10 @@ Role.prototype.isActionAllowed = function (requestObject, context, indexes) {
     return false;
   }
 
-  if (this.indexes._canCreate !== undefined &&
+  if (!doesIndexExist(requestObject, indexes) &&
+    this.indexes._canCreate !== undefined &&
     !this.indexes._canCreate &&
-    _.contains(['import', 'create', 'createCollection', 'putMapping', 'createOrUpdate'], requestObject.action) &&
-    !doesIndexExist(requestObject, indexes)) {
+    _.contains(['import', 'create', 'createCollection', 'putMapping', 'createOrUpdate'], requestObject.action)) {
```

Comment:
Indeed. Good news is: merging this will be easy as in this PR, I only moved the last test to the first position, to make it clearer that it handles the case when an index or a collection doesn't exist.